### ISSUE TITLE
Folding LoadAlignedVector* into the consumer instructions with VEX-encoding

### ISSUE THREAD

**Issue by fiigii** [dotnet/coreclr#21308]
Currently, `LoadVector128/256` can be folded into its consumer instructions with VEX-encoding but  `LoadAlignedVector128/256` not. 
`LoadAlignedVector128/256` would throw hardware exceptions if the memory address is not aligned to the specific boundary, but other VEX-encoded instructions (e.g., `vaddps xmm0, xmm1, [unalignedAddr]`) can work with unaligned memory. So, actually, we can fold `LoadVector128/256` into its consumer instructions with VEX-encoding.
```asm
;;; unoptimized
vmovaps xmm0, [unalignedAddr]  ;;; hardware exception
vaddps xmm0, xmm1, xmm0

;;; optimized
vaddps xmm0, xmm1, [unalignedAddr]  ;;; ok
```

All the mainstream C/C++ compilers have this behavior. 

@CarolEidt @tannergooding @mikedn

**Comment by tannergooding** [comment]
As mentioned in https://github.com/dotnet/corefx/issues/33566, we should not do this as it would silently remove an exception (`System.AccessViolationException`) that would otherwise have been thrown by the JIT.

**Comment by tannergooding** [comment]
I believe, instead, moving forward with a separate API that has the desired semantics and which still allows the end-user to use `LoadAligned` to enforce the alignment check is the best route.

**Comment by CarolEidt** [comment]
I agree that we don't want to silently remove an exception. It is not just that it violates the "strict" behavior that developers expect from the CLR, but could also cause sudden unexpected failures if something changed in the user code (or the optimization settings used) that caused the JIT to no longer fold the instruction.

The wide-ranging discussion in dotnet/corefx#33566 is very thought-provoking, and I propose that we continue the discussion for a bit longer. I don't know that we'll reach full consensus, but I'd like to get as many of the issues out there before we attempt to close on that one. And before we reach a decision on that, I'd like to see one or more of our API designers of long experience weigh in on the issue.

**Comment by CarolEidt** [comment]
@tannergooding @fiigii - We may want to change the title of this to indicate that this coreclr issue tracks the JIT implementation for  dotnet/corefx#33566

**Comment by fiigii** [comment]
> this coreclr issue tracks the JIT implementation for dotnet/corefx#33566

No, I think they are unrelated. I am proposing to fold `LoadAlignedVector*` with VEX-encoding, but dotnet/corefx#33566 wants to add new APIs.

**Comment by tannergooding** [comment]
I believe we had determined that folding `LoadAlignedVector*` under the VEX-encoding was not desirable because it would result in non-deterministic behavior (i.e. it would hide an `AccessViolation` that would otherwise be thrown for non-aligned inputs).
